superset/models/helpers.py

```diff
@@ -875,7 +875,9 @@ def mutate_query_from_config(self, sql: str) -> str:
         """Apply config's SQL_QUERY_MUTATOR
 
         Typically adds comments to the query with context"""
-        if sql_query_mutator := config["SQL_QUERY_MUTATOR"]:
+        if (sql_query_mutator := config.get("SQL_QUERY_MUTATOR")) and not config.get(
+            "MUTATE_AFTER_SPLIT"
```

Comment:
```suggestion
```

Reply:
@john-bodley I'm circling back to this old PR to see if I can get it merged. Do you think this statement needs to be removed? I copied it over from `superset/connectors/sqla/models.py` when drying up this code.